app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/repositories/PlaylistRepository.java

```diff
@@ -0,0 +1,126 @@
+package com.iitp.njack.iitp_connect.core.youtube.repositories;
+
+import android.arch.lifecycle.MutableLiveData;
+
+import com.google.api.client.extensions.android.http.AndroidHttp;
+import com.google.api.client.googleapis.extensions.android.gms.auth.GoogleAccountCredential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.model.Channel;
+import com.google.api.services.youtube.model.ChannelListResponse;
+import com.google.api.services.youtube.model.Playlist;
+import com.google.api.services.youtube.model.PlaylistListResponse;
+import com.iitp.njack.iitp_connect.core.youtube.models.YoutubePlaylist;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import io.reactivex.Observer;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+
+public class PlaylistRepository {
+    public GoogleAccountCredential googleAccountCredential;
+    private String channelName = "GoogleDevelopers";
+    private YouTube service;
+    private MutableLiveData<List<YoutubePlaylist>> playlists = new MutableLiveData<>();
+
+    @Inject
+    public PlaylistRepository(GoogleAccountCredential googleAccountCredential) {
+        this.googleAccountCredential = googleAccountCredential;
+        getDataFromAPI();
+    }
+
+    public void setGoogleAccountCredential(GoogleAccountCredential googleAccountCredential) {
+        this.googleAccountCredential = googleAccountCredential;
+    }
+
+    public MutableLiveData<List<YoutubePlaylist>> getPlaylists() {
+        return this.playlists;
+    }
+
+    public void getDataFromAPI() {
+        HttpTransport transport = AndroidHttp.newCompatibleTransport();
+        JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+        service = new YouTube.Builder(
```

Comment:
Use Dagger for it. Create an instance of YouTube builder and inject it in the Repository as a dependency just like `GoogleAccountCredential`.